File: user-service/src/main/java/br/com/userservice/core/document/Event.java
package br.com.userservice.core.document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Document(collection = "event")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Event {
        @Id
        private String id;
        private String userId;
        private String transactionId;
        private User payload;
        private String source;
        private String status;
        private List<History> eventHistory;
        private LocalDateTime createdAt;
}

File: user-registration/src/main/kotlin/br/com/userregistration/application/dto/Event.kt
package br.com.userregistration.application.dto

import br.com.userregistration.core.enums.ESagaStatus
import org.springframework.util.ObjectUtils
import java.time.LocalDateTime


class Event {
    private val id: String? = null
    private val userId: String? = null
    private val transactionId: String? = null
    private val payload: User? = null
    private var source: String? = null
    private var status: ESagaStatus? = null
    private var eventHistory: MutableList<History>? = null
    private val createdAt: LocalDateTime? = null

    fun addToHistory(history: History) {
        if (ObjectUtils.isEmpty(eventHistory)) {
            eventHistory = ArrayList()
        }
        eventHistory!!.add(history)
    }

    fun setStatus(status: ESagaStatus) { this.status = status};
    fun setSource(source: String) { this.source = source}
    fun getId(): String? {return this.id}
    fun getUserId(): String? { return this.userId};
    fun getTransactionId(): String? { return this.transactionId};
    fun getPayload(): User? { return this.payload};
    fun getSource(): String? { return this.source};
    fun getStatus(): ESagaStatus? { return this.status};
    fun getEventHistory(): MutableList<History>? {return this.eventHistory};
    fun getCreatedAt(): LocalDateTime? {return this.createdAt}

    override fun toString(): String {
        return "Event(id=$id, userId=$userId, transactionId=$transactionId, payload=$payload, source=$source, status=$status, eventHistory=$eventHistory, createdAt=$createdAt)"
    };
}